recipes/ffmpeg/all/conanfile.py

```diff
@@ -325,73 +322,61 @@ def validate(self):
 
     def build_requirements(self):
         if self.settings.arch in ("x86", "x86_64"):
-            self.build_requires("yasm/1.3.0")
-        self.build_requires("pkgconf/1.9.3")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+            self.tool_requires("yasm/1.3.0")
+        if not self.conf.get("tools.gnu:pkg_config", check_type=str):
+            self.tool_requires("pkgconf/1.9.3")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     @property
     def _target_arch(self):
-        triplet = tools.get_gnu_triplet(
-            "Macos" if is_apple_os(self) else str(self.settings.os),
-            str(self.settings.arch),
-            str(self.settings.compiler) if self.settings.os == "Windows" else None,
-        )
-        target_arch = triplet.split("-")[0]
-        return target_arch
+        # Taken from acceptable values https://github.com/FFmpeg/FFmpeg/blob/0684e58886881a998f1a7b510d73600ff1df2b90/configure#L5010
+        # All the values Conan's deffault setting support are already handled
+        return str(self.settings.arch)
```

Comment:
This does not map to the correct arch for arm64, at least on mac M1, probably on linux & android too.
It returns `armv8` instead of `aarch64`. This leads to trying to assemble arm 32-bit assembly which fails with things like:
```
src/libavutil/arm/bswap.h:59:31: warning: value size does not match register size specified by the constraint and modifier [-Wasm-operand-widths]
             : "+r"(x), "=&r"(t));
                              ^
src/libavutil/arm/bswap.h:55:19: note: use constraint modifier "w"
    __asm__ ("eor %1, %0, %0, ror #16 \n\t"
                  ^~
                  %w1
src/libavutil/arm/bswap.h:59:21: warning: value size does not match register size specified by the constraint and modifier [-Wasm-operand-widths]
             : "+r"(x), "=&r"(t));
                    ^
src/libavutil/arm/bswap.h:55:23: note: use constraint modifier "w"
    __asm__ ("eor %1, %0, %0, ror #16 \n\t"
                      ^~
                      %w0
```

Reply:
I'll try to fix that. @prince-chrismc please to not commit in this branch if you don't mind, it's a little bit confusing for me to see these modifications, and indeed all the arm stuff is likely broken.

Reply:
Sorry trying to be helpful! This is important ❤️ I put exception in for OS so I am not surprised we have a few here... just dont go crazy, if one is missing that's fine CCI only has a few limited support configs the rest are a bonus.

This is the code from 5.1 (and https://github.com/FFmpeg/FFmpeg/blob/n4.4.3/configure#L4892)

```
case "$arch" in
    aarch64|arm64)
        arch="aarch64"
    ;;
    arm*|iPad*|iPhone*)
        arch="arm"
    ;;
```

So `"armv8": "aarch64"` and probably the v7 32-bit as well?

Reply:
fixed in https://github.com/conan-io/conan-center-index/pull/15819/commits/fd1ed38fd5b65132ed379efb097a00ad2dd6d567